File: incanter/eza.clj
(ns default
  (:use (incanter zoo core stats charts io))
  (:use (clj-time [format :only (formatter formatters parse)] [coerce :only (to-long)]))
  )

;;(def url "http://www.oanda.com/currency/historical-rates/download?quote_currency=CNY&end_date=2012-9-3&start_date=2012-3-8&period=daily&display=absolute&rate=0&data_range=d180&price=bid&view=graph&base_currency_0=INR&base_currency_1=&base_currency_2=&base_currency_3=&base_currency_4=&download=csv")
;; Define the MSCI EZA Yahoo URL
(def ezaUrl "http://ichart.finance.yahoo.com/table.csv?s=EZA&ql=5&ignore=.csv")

;; Fetch the data
(def eza (read-dataset ezaUrl :header true))

;; Scatter Plot Graphs
(view (scatter-plot :High :Low :data eza))
(view (scatter-plot :Close (keyword "Adj Close") :data eza))

;; clj-time uses Joda Time
;; In this case we only want to use clj-time.format and clj-time.coerce
(defn dates-long "returns the dates as long" [eza]
  (let [ymd-formatter (formatters :year-month-day) dates-str ($ :Date eza)]
    (map #(to-long (parse ymd-formatter %)) dates-str))
  )

;; Creating 2 separate sets of data, Prices and Time
(def eza-prices ($ (keyword "Adj Close")  eza))
(def eza-times (dates-long eza))

;; Let's view the Simple Return plot
(view (time-series-plot eza-times eza-prices :x-label "Date" :y-label "EZA Simple Return"))

;; Calculating the Log Return Time Series
(def eza-pdf (roll-apply #(apply - (log %)) 2 eza-prices))

;; Let's view the Log Return plot
(view (time-series-plot eza-times eza-pdf :x-label "Date" :y-label "EZA Log Return"))


(view eza)
(view eza-times)
(view eza-prices)
(view eza-pdf)

(log (/ 65.04 65.72))

File: R_Code/eza.R
# eza.R  		script file for MSCI EZA comparison with Incanter
#
# author: Setra R
# created: September 18, 2013
# revised: 

# Incanter displays 16 digits
options(digits=16)

# Let's play with MSCI EZA from Yahoo
url <-"http://ichart.finance.yahoo.com/table.csv?s=EZA&a=01&b=01&c=2010&d=01&e=01&f=2013&g=d&ignore=.csv"
eza <- read.csv(url, colClasses=c('Date'='Date'))

# Let's display some data
plot(eza$Date,eza$Close)

# Close Field will be our reference
eza.prices <- as.ts(eza$Adj.Close)

# Log Return Calculation
eza.pdf <- log(eza.prices)-log(lag(eza.prices))

# Descriptive Statistics
head(eza.pdf)     # 0.01996134628454538  0.01306806205664746 -0.01199829806680075  0.01122484300465132 -0.02035732656792355
mean(eza.pdf)     # 0.0004099698056707165
var(eza.pdf)      # 0.0003574576005939714
sd(eza.pdf)       # 0.01890654914557311

library(PerformanceAnalytics)
skewness(eza.pdf) # -0.3196135749374513
kurtosis(eza.pdf) # 2.423887802277989

# Displays the histogram
hist(eza.pdf)

require(zoo)
require(forecast) 
eza.times <- as.Date(eza$Date,"%Y-%b-%d") 
eza.z <- zoo(eza.prices,eza.times)
plot(eza.z, xlab="Date", ylab="EZA Simple Returns")

eza.r <- zoo(eza.pdf,eza.times)
plot(eza.r, xlab="Date", ylab="EZA Log Returns")